WatchFaceKotlin/wear/src/main/java/com/example/android/wearable/watchfacekotlin/watchface/AnalogComplicationWatchFaceService.kt

```diff
@@ -0,0 +1,834 @@
+            backgroundColor = sharedPreferences.getInt(backgroundColorResourceName, Color.BLACK)
+            val markerColorResourceName = applicationContext.getString(R.string.saved_marker_color)
+
+            // Set defaults for colors
+            watchHandHighlightColor = sharedPreferences.getInt(markerColorResourceName, Color.RED)
+            if (backgroundColor == Color.WHITE) {
+                watchHandAndComplicationsColor = Color.BLACK
+                watchHandShadowColor = Color.WHITE
+            } else {
+                watchHandAndComplicationsColor = Color.WHITE
+                watchHandShadowColor = Color.BLACK
+            }
+            val unreadNotificationPreferenceResourceName =
+                applicationContext.getString(R.string.saved_unread_notifications_pref)
+            unreadNotificationsPreference =
+                sharedPreferences.getBoolean(unreadNotificationPreferenceResourceName, true)
+        }
+
+        private fun initializeComplicationsAndBackground() {
+            Log.d(TAG, "initializeComplications()")
+
+            // Initialize background color (in case background complication is inactive).
+            backgroundPaint = Paint()
+            backgroundPaint.color = backgroundColor
+
+            // Creates a ComplicationDrawable for each location where the user can render a
+            // complication on the watch face. In this watch face, we create one for left, right,
+            // and background, but you could add many more.
+            val leftComplicationDrawable = ComplicationDrawable(applicationContext)
+            val rightComplicationDrawable = ComplicationDrawable(applicationContext)
+            val backgroundComplicationDrawable = ComplicationDrawable(applicationContext)
+
+            // Adds new complications to a SparseArray to simplify setting styles and ambient
+            // properties for all complications, i.e., iterate over them all.
+            complicationDrawableSparseArray = SparseArray(complicationIds.size)
+            complicationDrawableSparseArray.put(LEFT_COMPLICATION_ID, leftComplicationDrawable)
+            complicationDrawableSparseArray.put(RIGHT_COMPLICATION_ID, rightComplicationDrawable)
+            complicationDrawableSparseArray.put(
+                BACKGROUND_COMPLICATION_ID,
+                backgroundComplicationDrawable
+            )
+            setComplicationsActiveAndAmbientColors(watchHandHighlightColor)
+            setActiveComplications(*complicationIds)
+        }
+
+        private fun initializeWatchFace() {
+            hourPaint = Paint()
+            hourPaint.color = watchHandAndComplicationsColor
+            hourPaint.strokeWidth = HOUR_STROKE_WIDTH
+            hourPaint.isAntiAlias = true
+            hourPaint.strokeCap = Paint.Cap.ROUND
+            hourPaint.setShadowLayer(
+                SHADOW_RADIUS.toFloat(),
+                0f,
+                0f,
+                watchHandShadowColor
+            )
+
```

Comment:
You can use the `apply` method here to improve readability:
```
            hourPaint = Paint().apply {
                color = watchHandAndComplicationsColor
                strokeWidth = HOUR_STROKE_WIDTH
                isAntiAlias = true
                strokeCap = Paint.Cap.ROUND
                setShadowLayer(
                    SHADOW_RADIUS.toFloat(),
                    0f,
                    0f,
                    watchHandShadowColor
                )
            }
```

Reply:
Done and applied to others below.